projects/portal/src/app/pages/delegate/validators/validators.usecase.service.ts

```diff
@@ -0,0 +1,134 @@
+import { CosmosRestService } from '../../../models/cosmos-rest.service';
+import { StoredWallet } from '../../../models/wallets/wallet.model';
+import { WalletService } from '../../../models/wallets/wallet.service';
+import { validatorType } from '../../../views/delegate/validators/validators.component';
+import { Injectable } from '@angular/core';
+import cosmosclient from '@cosmos-client/core';
+import {
+  InlineResponse20038DelegationResponses,
+  InlineResponse20041Validators,
+  InlineResponse20047,
+} from '@cosmos-client/core/esm/openapi';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { concatMap, filter, map, mergeMap, withLatestFrom } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ValidatorsUseCaseService {
+  constructor(
+    private readonly walletService: WalletService,
+    private readonly cosmosRest: CosmosRestService,
+  ) {}
+  get currentStoredWallet$(): Observable<StoredWallet | null | undefined> {
+    return this.walletService.currentStoredWallet$;
+  }
+  get validatorsList$(): Observable<InlineResponse20041Validators[] | undefined> {
+    return this.cosmosRest.getValidators$();
+  }
+  get accAddress$(): Observable<cosmosclient.AccAddress> {
+    return this.walletService.currentStoredWallet$.pipe(
+      filter((wallet): wallet is StoredWallet => wallet !== undefined && wallet !== null),
+      map((wallet) => cosmosclient.AccAddress.fromString(wallet.address)),
+    );
+  }
```

Comment:
In case of anomaly, no value is returned due to filter. (Observable does not pass any values) How can I test it?

Reply:
Since it is sufficient to guarantee that "values do not flow", it is better to write ` throw new Error('Not reached')`, for example.
```ts
   service.accAddress$().subscribe({
    next: () => {
      throw new Error('Not reached');
    },
    complete: () => done(),
  });
```

However, if it is a hot observable, it may not work because it is not complete.

It may be possible to test using "marble test", but unfortunately I'm not familiar with it either

Reply:
Thank you. I think it is "Hot".  because, it is never finish. 

I reserched about "marble test"
[https://mokkapps.de/blog/how-i-write-marble-tests-for-rxjs-observables-in-angular/](https://mokkapps.de/blog/how-i-write-marble-tests-for-rxjs-observables-in-angular/)

I installed "jest-marble" just a little bit,  but it required 'expect/build/jasmineUtils' so, I couldn't do it
[https://www.npmjs.com/package/jest-marbles](https://www.npmjs.com/package/jest-marbles)

For Observables with filters, I think it's enough to test only the normal case.